### PR TITLE
refactor(core): refactoring to achieve platform agnostic

### PR DESCRIPTION
**Pull request by nikolovlazar**
Fixes: https://github.com/getsentry/sentry-react-native/issues/1966

- Removed the browser dependencies (both sentry and fullstory)
- Added only the `@sentry/core` dependency
- Made the client option required, so either the React Native or Web FullStory client can be passed from the user-land

**BREAKING:** The client option is now required.

If this implementation is reviewed and accepted, I'll update the README to reflect on the latest breaking changes and produce a simple Migration documentation.

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
After this gets merged/released we might need to update this page too.

https://partners.fullstory.com/integrations/solution/780/sentry

Also, we should update the docs in [README.md](https://github.com/getsentry/sentry-fullstory/tree/9a3eb5d3e8c3d84bdc628be27c5b8e63c2272855#readme)

File: src/SentryFullStory.ts
Comment (AbhiPrasad) [inline src/SentryFullStory.ts:35]: You might not even need `@sentry/core` since `addGlobalEventProcessor` and `getCurrentHub` are dependency injected into `setupOnce`.

See: https://github.com/getsentry/sentry-javascript/blob/2d068e9f944a96dd566519a9e38ae05fc7b5b769/packages/types/src/integration.ts#L25

File: CHANGELOG.md
Comment (AbhiPrasad) [inline CHANGELOG.md:7]: So this will be `2.0.0` then?  Can we add a short migration snippet for users here?

We probably then need to update the README as well!

File: package.json
Comment (AbhiPrasad) [inline package.json:26]: What do you think about adding a Sentry peer dep of `>=4 and <=7`? This way if a new major version of the Sentry SDKs breaks the integrations API people won't get this integration

We can peerdep `@sentry/core`, since that should be required by both react native and browser (and all other possible packages).

**Review by AbhiPrasad — APPROVED** [review]

**Comment by marandaneto** [comment]
@nikolovlazar can we unblock and release this? :) cc @krystofwoldrich

File: package.json
Comment (krystofwoldrich) [inline package.json:26]:
```suggestion
  "peerDependencies": {
  	"@sentry/core": "4.x || 5.x || 6.x || 7.x"
  },
```

**Comment by b13i** [comment]
Anything remaining to get this merged? Was checking in on the root issue https://github.com/getsentry/sentry-react-native/issues/1966 and got excited to see the PR is open + approved 😄

File: CHANGELOG.md
Comment (krystofwoldrich) [reply CHANGELOG.md:7]: I've updated the readme so we can merge this.

**Comment by krystofwoldrich** [comment]
@b13i The project readme and changelog need an update.

**Pull request merged**
